### PR TITLE
ENT-8890: Added utility function to get time classes

### PR DESCRIPTION
**Pull request by larsewi**
With the GetTimeClasses() function you can get a set of time classes
without the need for an EvalContext object.

Ticket: ENT-8890
Changelog: None
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>

[Build Status](https://ci.cfengine.com/job/pr-pipeline/10835/)

### CODE REVIEW

**Comment by larsewi** [comment]
> [Build Status](https://ci.cfengine.com/job/pr-pipeline/10835/)

^ Sequential test failure is unrelated

**Review by vpodzime — APPROVED** [review]
Yes! Very nice! :clap: :+1:

File: libenv/time_classes.h [hunk 1]
```diff
@@ -27,6 +27,14 @@
 
 #include <cf3.defs.h>
 
+/**
+ * @brief Get a string set of time classes based on a timestamp.
+ * @param time Unix timestamp
+ * @return Set of time classes or NULL in case of error.
```
Comment (olehermanse) [inline libenv/time_classes.h:33]: A couple of examples would be nice here.

**Review by olehermanse — APPROVED** [review]
👍

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10845/)

File: libpromises/eval_context.h [hunk 2]
```diff
@@ -254,6 +254,14 @@ StringSet *ClassesMatchingLocal(const EvalContext *ctx, const char* regex, const
 bool EvalProcessResult(const char *process_result, StringSet *proc_attr);
 bool EvalFileResult(const char *file_result, StringSet *leaf_attr);
 
+/**
+ * @brief Evaluates a class expression based a set of defined classes.
```
Comment (vpodzime) [inline libpromises/eval_context.h:258]: *based **on** a set*

**Review by vpodzime — APPROVED** [review]
Just a nitpick. Really nice, otherwise! You were lucky :grin:

**Review by vpodzime — APPROVED** [review]
ACK

**Pull request merged**